src/overlays/actors/ovl_Obj_Y2lift/z_obj_y2lift.c

```diff
@@ -28,25 +28,72 @@ const ActorInit Obj_Y2lift_InitVars = {
     (ActorFunc)ObjY2lift_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B9A940[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 800, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 800, ICHAIN_STOP),
 };
 
-#endif
+void ObjY2lift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjY2lift* this = THIS;
 
-extern InitChainEntry D_80B9A940[];
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 1);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_kaizoku_obj_Colheader_0019B0);
+}
 
-extern UNK_TYPE D_06001680;
-extern UNK_TYPE D_060019B0;
+void ObjY2lift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjY2lift* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Y2lift/ObjY2lift_Init.s")
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Y2lift/ObjY2lift_Destroy.s")
+void ObjY2lift_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 temp_fv0;
+    f32 sp1C;
+    s32 temp_v0;
+    ObjY2lift* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Y2lift/ObjY2lift_Update.s")
+    temp_fv0 = this->dyna.actor.world.pos.y;
+    sp1C = 0.0f;
+    temp_v0 = DynaPolyActor_IsInRidingMovingState(&this->dyna);
```

Comment:
```suggestion
    ObjY2lift *this = THIS;
    f32 temp_fv0 = this->dyna.actor.world.pos.y;
    f32 sp1C = 0.0f;
    s32 temp_v0 = DynaPolyActor_IsInRidingMovingState(&this->dyna);

```
looks much better.

Reply:
it doesn't match

Reply:
https://decomp.me/scratch/apIjs
Strange, it did in the scratch.

Reply:
And locally.

Reply:
you're right, i forgot to remove the pad. Fixed